### PR TITLE
fix error opening files with special characters on Linux

### PR DESCRIPTION
**Pull request by marcusgregory**
#150

### CODE REVIEW

File: lib/src/plaform/open_file.dart [hunk 1]
```diff
@@ -26,10 +26,13 @@ class OpenFile {
       if (Platform.isMacOS) {
         _result = mac.system(['open', '$filePath']);
       } else if (Platform.isLinux) {
+        var filePathLinux = Uri.file(filePath!);
```
Comment (mx1up) [inline lib/src/plaform/open_file.dart:29]: this is the real fix for the special chars and spaces. it does assume that the linux.system call supports file uri's (xdg-open definitely does). I assume @marcusgregory has tested this and can validate this. I'll also test this

File: lib/src/plaform/open_file.dart [hunk 2]
```diff
@@ -26,10 +26,13 @@ class OpenFile {
...
         if (linuxByProcess) {
-          _result = Process.runSync('xdg-open', [filePath!]).exitCode;
+          _result = Process.runSync('xdg-open', ['${filePathLinux.toString()}'])
```
Comment (mx1up) [inline lib/src/plaform/open_file.dart:31]: no string interpolation needed

File: lib/src/plaform/open_file.dart [hunk 3]
```diff
@@ -26,10 +26,13 @@ class OpenFile {
...
+              .exitCode;
         } else {
-          _result = linux.system(['$linuxDesktopName-open', '$filePath']);
+          _result = linux.system(
+              ['$linuxDesktopName-open', '"${filePathLinux.toString()}"']);
```
Comment (mx1up) [inline lib/src/plaform/open_file.dart:35]: no string interpolation needed

File: lib/src/plaform/open_file.dart (hunk 2 above)
Comment (marcusgregory) [reply lib/src/plaform/open_file.dart:31]: @mx1up it's not really necessary, I just used it to make it easier to understand.

**Pull request merged**